src/src/Helpers/Networking.cpp

```diff
@@ -1544,7 +1544,26 @@ int http_authenticate(const String& logIdentifier,
     event += '=';
     event += httpCode;
     eventQueue.addMove(std::move(event));
+    #if FEATURE_THINGSPEAK_EVENT
+      // Generate event with the response of a 
+      // one value one channel tingspeak request
+      // e.g. (sendToHTTP,api.thingspeak.com,80,/channels/143789/fields/5/last)
+      // where first eventvalue is the channel number, the second the field number
+      // and the third is the value received by the request
+      if (equals(host, F("api.thingspeak.com")) && equals(uri.substring(uri.length() - 5, uri.length()), F("/last")) && httpCode == 200) {
+      String revent = F("reply#");
+      revent += host;
+      revent += '=';
+      revent += uri.substring(0, uri.length() - 14).substring(10, uri.length()); //get the channel number
```

Comment:
Looks like you try to do this:
```c++
revent += uri.substring(10, uri.length() - 14); //get the channel number
```

Reply:
no.... to get the channel number i have to start at the end since the channel number is the only value in the uri, that has an unknown length. so `/channels/143789/fields/5/last` gets `/channels/143789` and after that i  cut off the first part since it is also static

Reply:
ah wait.. maybe you are right.. 🫣

Reply:
Since you all have such nice separation by '/', why not use those markers to get a proper substring?
You can use the function `parseString` from "Helpers/StringConverter.h"
```c++
parseString(uri.c_str(), 3, '/');
```
You may need to double check whether the index 3 is correct here...